cpp/include/raft/linalg/gemm.cuh

```diff
@@ -27,8 +28,65 @@
 #include <raft/core/resources.hpp>
 #include <raft/util/input_validation.hpp>
 
-namespace raft {
-namespace linalg {
+namespace raft::linalg {
```

Comment:
I wonder if instead of gemm.hpp, we should be putting the new matmul function in matmul.hpp?